lib/libc/stdlib/malloc/snmalloc/Makefile.inc

```diff
@@ -0,0 +1,48 @@
+.PATH: ${LIBC_SRCTOP}/stdlib/malloc/snmalloc
+
+MISRCS+=	malloc.cc
+
+# Specify the locations in contrib for headers.
+CXXFLAGS+=	-I${SRCTOP}/contrib/subrepo-snmalloc/src/snmalloc
+
+CXXSTD?=	c++20
+
+# C++ dialect options.
+CXXFLAGS+=	-fno-exceptions -fno-rtti
+.if ${MACHINE_ARCH} == amd64
+# -mcx16 is needed to avoid calls to libatomic.
+CXXFLAGS+=	-mcx16
+.endif
+
+# Require an explicit call from libthr to clean up thread-local state
+CXXFLAGS+=	-DSNMALLOC_USE_THREAD_CLEANUP
+# Export the __je_bootstrap_* functions for TLS allocation.
```

Comment:
`__je_bootstrap_*`, really? Surely these can be given a more sensible name...

Reply:
Hmm, I think we could actually drop that bit for CheriBSD. We're using using a copy of tls_malloc for TLS rather than hooking into the system allocator. Upstream should probably do the same, but likely should use rtld_malloc.c.